Mathlib/Data/Sum/Basic.lean

```diff
@@ -201,6 +202,75 @@ section LiftRel
 #align sum.lift_rel.swap Sum.LiftRel.swap
 #align sum.lift_rel_swap_iff Sum.liftRel_swap_iff
 
+mk_iff_of_inductive_prop Sum.LiftRel Sum.liftRel_iff
+
+variable {r : α → γ → Prop} {s : β → δ → Prop} {x : Sum α β} {y : Sum γ δ}
+  {a : α} {b : β} {c : γ} {d : δ}
+
+theorem isLeft_eq_of_liftRel (h : LiftRel r s x y) : x.isLeft = y.isLeft := by
+  cases h <;> rfl
+
+theorem isRight_eq_of_liftRel (h : LiftRel r s x y) : x.isRight = y.isRight := by
+  cases h <;> rfl
+
+theorem isLeft_eq_of_liftRel_inl_fst (h : LiftRel r s (inl a) y) : y.isLeft :=
+  (isLeft_eq_of_liftRel h).symm
+
+theorem isLeft_eq_of_liftRel_inl_snd (h : LiftRel r s x (inl c)) : x.isLeft :=
+  isLeft_eq_of_liftRel h
+
+theorem isRight_of_liftRel_inr_fst (h : LiftRel r s (inr b) y) : y.isRight :=
+  (isRight_eq_of_liftRel h).symm
+
+theorem isRight_of_liftRel_inr_snd (h : LiftRel r s x (inr d)) : x.isRight :=
+  isRight_eq_of_liftRel h
+
+variable {τ κ : Type*} {f : τ → Sum α β} {g : κ → Sum γ δ} {t : τ} {k : κ}
+
+theorem isLeft_apply_of_inl_fst (h : LiftRel r s (inl a) (g k)) : (g k).isLeft :=
+  isLeft_eq_of_liftRel_inl_fst h
+
+theorem isLeft_apply_of_inl_snd (h : LiftRel r s (f t) (inl c)) : (f t).isLeft :=
+  isLeft_eq_of_liftRel_inl_snd h
+
+theorem isRight_apply_of_inr_fst (h : LiftRel r s (inr b) (g k)) : (g k).isRight :=
+  isRight_of_liftRel_inr_fst h
+
+theorem isRight_apply_of_inr_snd (h : LiftRel r s (f t) (inr d)) : (f t).isRight :=
+  isRight_of_liftRel_inr_snd h
+
+theorem isLeft_apply_eq_of_LiftRel_apply (h : LiftRel r s (f t) y) : (f t).isLeft = y.isLeft :=
+  isLeft_eq_of_liftRel h
+
+theorem isRight_apply_eq_of_LiftRel_apply (h : LiftRel r s x (g k)) :
+    (g k).isRight = x.isRight := (isRight_eq_of_liftRel h).symm
```

Comment:
Sorry, why do we need these? Are they not literally syntactical specialisations of the previous?

Reply:
You'd think that, I certainly did, but my recollection for when I was working on this back in September is that for some reason the typechecker couldn't prove things of this form using those lemmas. I can see if that has changed though.

Reply:
I would be highly surprised if that were the case! I can understand the difference between having your first set of lemmas and not having any lemma, but I can't imagine that your second set of lemmas gives you anything more than your first set.

Reply:
You are right! Removed.